Yield compile failed with C2589 in msvc

### Search before asking

- [ ✔ ] I searched the [issues](https://github.com/alibaba/async_simple/issues) and found no similar issues.


### What happened + What you expected to happen

```cpp
co_await async_simple::coro::Yield(); // compile failed 
```

output:
```sh
error C2589: ';': illegal token on right side of '::'
```

### Reproduction way

Visual Studio 2022

### Anything else

May be we should use `{}` instead of `()` , it can work in msvc.
```cpp
co_await async_simple::coro::Yield{}; 
```

### Are you willing to submit a PR?

- [ ✔ ] Yes I am willing to submit a PR!

--- a/async_simple/coro/Collect.h
+++ b/async_simple/coro/Collect.h
@@ -380,7 +380,7 @@ inline auto collectAllWindowedImpl(size_t maxConcurrency,
             output.push_back(std::move(t));
         }
         if (yield) {
-            co_await Yield();
+            co_await Yield{};
         }
     }
     co_return std::move(output);

--- a/async_simple/coro/Lazy.h
+++ b/async_simple/coro/Lazy.h
@@ -42,7 +42,7 @@ class Lazy;
 // execute, we could write:
 //
 // ```C++
-//  co_await Yield();
+//  co_await Yield{};
 // ```
 //
 // This would suspend the executing coroutine.

--- a/docs/docs.cn/Lazy.md
+++ b/docs/docs.cn/Lazy.md
@@ -291,7 +291,7 @@ RescheduleLazy和Lazy持有的LazyLocals为一个void*类型的指针，类型�
 # Yield
 
 有时我们可能希望主动让出一个 Lazy 协程的执行，将执行资源交给其他任务。（例如我们发现当前协程执行的时间太长了）
-我们可以通过在 Lazy 中 `co_await async_simple::coro::Yield();` 来做到这一点。
+我们可以通过在 Lazy 中 `co_await async_simple::coro::Yield{};` 来做到这一点。
 
 # 获取当前调度器
 

--- a/docs/docs.en/Lazy.md
+++ b/docs/docs.en/Lazy.md
@@ -290,7 +290,7 @@ If the user starts Lazy through 'start (callback)' and needs to recycle LazyLoca
 # Yield
 
 Sometimes we may want the executing Lazy to yield out. (For example, we found the Lazy has been executed for a long time)
-We can yield it by `co_await async_simple::coro::Yield();` in the Lazy.
+We can yield it by `co_await async_simple::coro::Yield{};` in the Lazy.
 
 # Get the Current Executor
 

